docs/sovereign/standalone-evm.md

```diff
@@ -0,0 +1,172 @@
+
+### 6.2 Token Storage in EVM
+
+Token balances (like ERC20) live in the contract’s own storage. The contract will use the last 20 bytes of a user’s MvX address when recording ownership or balances. If an opcode like `GetCaller` is executed, it returns only the last 20 bytes from the `ContractCallInput.Sender`.
+
+### 6.3 Calling WasmVM from EVM
+
+MultiversX WasmVM expects 32-byte addresses. If an EVM contract tries to invoke a WasmVM contract using only 20 bytes, the call will fail due to incorrect argument size. Consequently, when the EVM calls a WasmVM contract, it must supply a full 32-byte address.
+
+:::note
+In most cases, the EVM contracts will call only other EVM contracts. However, bridging to WasmVM is still feasible, for example, when claiming ESDT tokens through an ERC wrapper contract.
+:::
+
+---
+
+## 7. WASM VM and the `ExecuteOnDestOnOtherVM` Function
+
+The **WASM VM** supports a public function `ExecuteOnDestOnOtherVM` via the **BlockchainHook** interface. If a new VM is fully integrated, it can be added to the `vmContainer` component with a new **baseAddress**. Below is an example table illustrating potential base addresses for different VMs:
+
+| VM Name     | Example Base Address | Notes                                                                            |
+|-------------|----------------------|----------------------------------------------------------------------------------|
+| **WASM VM** | 05                   | Standard base address for the WASM VM                                            |
+| **System VM** | 255                | Standard base address (example) for the System VM                                |
+| **EVM**     | To Be Decided        | Will be assigned upon integration to ensure address derivation works properly    |
+
+From the `SCAddress`, the protocol looks at bytes **10** and **11** to determine which VM should be called. Once EVM integration is complete, it will receive its own base address and will adjust how the **CreateContract** opcode calculates deployed contract addresses.
+
+### 7.1 Synchronous Execution
+
+When the EVM executes a `DelegateCall` opcode, it will invoke an internal function of the new EVM implementation that checks whether execution should occur in the EVM itself or a different VM. If it needs to run on another VM, it calls `blockchainHook.ExecuteOnDestOnOtherVM`.
+
+- **Returning `VMOutput`**: This function returns a `VMOutput`, which can be merged into the current `outputContext` and `storageContext` via the `PushContext`-type public functions.
+
+In the **WASM VM**, if a smart contract calls `ExecuteOnDest`, the VM decides where the execution should take place. For asynchronous calls, the same logic applies:
+
+- **Intra-Shard**: The system calls `ExecuteOnDestOnOtherVM`.  
+- **Cross-Shard**: On the destination shard, the **scProcessor** determines which VM to invoke and continues accordingly.
+
+---
+
+## 10. ESDT ↔ ERC20 & ESDTNFT ↔ ERC721
```

Comment:
No other sections between `# 7` and `# 10`. Maybe drop the numbering in all headers?

Reply:
true, I had some other details, considered unnecessary afterwards.